torch/csrc/Module.cpp

```diff
@@ -565,6 +566,16 @@ static void warning_handler(
   }
 }
 
+// In Python we can't use the trick of C10_LOG_API_USAGE_ONCE
+// Guaranteed to be invoked from Python under GIL, no locking on map needed
+static void LogAPIUsageOnceFromPython(const std::string& event) {
+  static std::unordered_set<std::string> seen;
+  if (!seen.count(event)) {
```

Comment:
This looks uber slow lol

Reply:
it's called from python, so it should be ok :) I'm putting it only in construction of classes, so the impact should be minimal